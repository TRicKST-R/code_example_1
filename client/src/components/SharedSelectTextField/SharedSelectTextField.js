import React, { Component } from 'react'
import TextField from 'material-ui/TextField'
import MenuItem from 'material-ui/Menu/MenuItem'
import PropTypes from 'prop-types'

class SharedSelectTextField extends Component {
  render () {
    const { data = [], label, inputStyles, handleChange, value } = this.props
    return (
      <TextField
        id='select-currency'
        fullWidth
        select
        label={label}
        value={value}
        disabled={data.length === 0}
        margin='normal'
        type='text'
        onChange={handleChange}
        InputLabelProps={{
          shrink: true,
          className: inputStyles
        }}
        SelectProps={{
          MenuProps: {
            PaperProps: {
              style: {
                /* height of element(48px * 4) + 16px of list padding */
                maxHeight: 208
              }
            }
          }
        }}
      >
        {data.map((option, i) => (
          <MenuItem title={option.label} key={i} value={option.value}>
            {option.label}
          </MenuItem>
        ))}
      </TextField>
    )
  }
}

SharedSelectTextField.propTypes = {
  /* name */
  label: PropTypes.string,
  /* selected value */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /* array of options with shape {label, value} */
  data: PropTypes.array,
  /* styles */
  inputStyles: PropTypes.string,
  /* listen to user input and changes state of parent component  */
  handleChange: PropTypes.func
}

export default SharedSelectTextField
