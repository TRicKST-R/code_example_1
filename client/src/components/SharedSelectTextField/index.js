import SharedSelectTextField from './SharedSelectTextField'

export default SharedSelectTextField
