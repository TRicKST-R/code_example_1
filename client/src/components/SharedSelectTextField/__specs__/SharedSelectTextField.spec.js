import React from 'react'
import { shallow } from 'enzyme'
import SharedSelectTextField from '../SharedSelectTextField'

describe('<SharedSelectTextField />', () => {
  let wrapper, data, label, inputStyles, handleChange, value
  beforeEach(() => {
    data = [
      {
        value: 'value',
        label: 'label'
      }
    ]
    label = 'label'
    value = 'value'
    inputStyles = 'inputStyles'
    handleChange = jest.fn()
    wrapper = shallow(
      <SharedSelectTextField
        data={data}
        label={label}
        value={value}
        inputStyles={inputStyles}
        handleChange={handleChange}
      />
    )
  })
  it('should renders as expected', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
