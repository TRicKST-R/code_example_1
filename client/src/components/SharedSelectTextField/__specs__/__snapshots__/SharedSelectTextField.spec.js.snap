// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SharedSelectTextField /> should renders as expected 1`] = `
<TextField
  InputLabelProps={
    Object {
      "className": "inputStyles",
      "shrink": true,
    }
  }
  SelectProps={
    Object {
      "MenuProps": Object {
        "PaperProps": Object {
          "style": Object {
            "maxHeight": 208,
          },
        },
      },
    }
  }
  disabled={false}
  fullWidth={true}
  id="select-currency"
  label="label"
  margin="normal"
  onChange={[Function]}
  required={false}
  select={true}
  type="text"
  value="value"
>
  <withStyles(MenuItem)
    key="0"
    value="value"
  >
    label
  </withStyles(MenuItem)>
</TextField>
`;
